pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/partition/PartitionFunctionFactory.java

```diff
@@ -60,13 +60,15 @@ private PartitionFunctionFactory() {
    *
    * @param functionName Name of partition function
    * @param numPartitions Number of partitions.
+   * @param functionConfig The function configuration for given function.
    * @return Partition function
    */
   // TODO: introduce a way to inject custom partition function
   // a custom partition function could be used in the realtime stream partitioning or offline segment partitioning.
   // The PartitionFunctionFactory should be able to support these default implementations, as well as instantiate
   // based on config
-  public static PartitionFunction getPartitionFunction(String functionName, int numPartitions) {
+  public static PartitionFunction getPartitionFunction(String functionName, int numPartitions,
+                                                       Map<String, String> functionConfig) {
```

Comment:
Please reformat the file with [Pinot Style](https://docs.pinot.apache.org/developers/developers-and-contributors/code-setup#intellij)
```suggestion
      @Nullable Map<String, String> functionConfig) {
```

Reply:
done.